### PR TITLE
[SBOM] add `dd_env` to the sbom payload

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
Adds `DD_ENV` to the SBOMPayload

### Motivation
Differentiate vulnerability present in a PROD env from a staging, test env...

### Additional Notes
This [PR](https://github.com/DataDog/agent-payload/pull/238) needs to be merged first

### Describe how to test/QA your changes
Use the right protobuf definitions with:
`go get github.com/DataDog/agent-payload/v5@4f1f6cf94882983c66c2013a7c7a230e7a5fc3b0`

Add a log line to print the SBOM Payload and build the new binary.

Deploy with

```
datadog:
  env:
    - name: DD_CONTAINER_IMAGE_COLLECTION_METADATA_ENABLED
      value: "true"
    - name: DD_CONTAINER_IMAGE_COLLECTION_SBOM_ENABLED
      value: "true"
    - name: DD_CONTAINER_IMAGE_ENABLED
      value: "true"
    - name: DD_SBOM_ENABLED
      value: "true"
    - name: DD_ENV
      value: my-env
  confd:
    container_image.yaml: |
      init_config:
      instances:
        - chunk_size: 1
          new_images_max_latency_seconds: 10
          periodic_refresh_seconds: 60
    sbom.yaml: |
      init_config:
      instances:
        - chunk_size: 1
          new_sbom_max_latency_seconds: 10
          periodic_refresh_seconds: 60
```

Check the logs with `k logs pod | grep -i <YOUR_LOG>`.

Here are some log lignes obtained by logging `SBOMPayload.GetDdEnv()`:
```
2023-03-14 20:33:23 UTC | CORE | INFO | (pkg/collector/corechecks/sbom/processor.go:49 in func1) | SBOMPayload: %v my-env
2023-03-14 20:33:23 UTC | CORE | INFO | (pkg/collector/corechecks/sbom/processor.go:49 in func1) | SBOMPayload: %v my-env

```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/add-dd_env-sbompayload.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+upgrade:
+enhancements:
+  - |
+    add `DD_ENV` to the SBOMPayload in the SBOM check
```
Comment (kayayarai) [inline releasenotes/notes/add-dd_env-sbompayload.yaml:5]:
```suggestion
    Added `DD_ENV` to the SBOMPayload in the SBOM check.
```

**Review by kayayarai — APPROVED** [review]
small edit to release note

File: pkg/collector/corechecks/sbom/processor.go [hunk 2]
```diff
@@ -24,6 +25,7 @@ import (
 )
 
 var /* const */ (
+	envVarEnv   = ddConfig.Datadog.GetString("env")
```
[the remark this answers is not in the document]
Comment (AliDatadog) [reply pkg/collector/corechecks/sbom/processor.go:28]: I am not sure of the exact mechanism either but I saw similar pattern in [other places](https://github.com/DataDog/datadog-agent/blob/e7cda93e024fee866b04eed308c82b402aa9d9a4/cmd/agent/gui/checks.go#L34). Also, I could see the correct value during tests

[the remark this answers is not in the document]
Comment (AliDatadog) [reply pkg/collector/corechecks/sbom/processor.go:28]: I think that `GetString` will parse the config if needed.

**Review by L3n41c — APPROVED** [review]

**Pull request merged**